frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx

```diff
@@ -14,36 +14,36 @@ import { getSignedPreviewUrl, getUnsignedPreviewUrl, getSignedToken } from "meta
 import { getSiteUrl, getEmbeddingSecretKey, getIsPublicSharingEnabled, getIsApplicationEmbeddingEnabled } from "metabase/selectors/settings";
 import { getUserIsAdmin } from "metabase/selectors/user";
 
-import type { Parameter, ParameterId } from "metabase/meta/types/Parameter";
-
 import MetabaseAnalytics from "metabase/lib/analytics";
 
+import type { Parameter, ParameterId } from "metabase/meta/types/Parameter";
+import type { EmbeddableResource, EmbeddingParams } from "metabase/public/lib/types";
+
 export type Pane = "preview"|"code";
 export type EmbedType = null|"simple"|"application";
 
-export type EmbeddingParams = {
-    [key: string]: string
-}
-
 export type DisplayOptions = {
     theme: ?string,
     bordered: boolean,
     titled: boolean,
 }
 
-export type EmbeddableResource = {
-    id: string,
-    public_uuid: string,
-    embedding_params: EmbeddingParams
-}
-
 type Props = {
     className?: string,
-    siteUrl: string,
-    secretKey: string,
     resource: EmbeddableResource,
     resourceType: string,
     resourceParameters: Parameter[],
+
+    isAdmin: boolean,
+    siteUrl: string,
+    secretKey: string,
+
+    // Flow doesn't understand these are provided by @connect?
```

Comment:
Seems that Flow doesn't understand object decorators. By default it warns about them, but we have suppressed the warnings in `.flowconfig`. So probably the right way to fix this is to use the old-school `export default connect(...)(...);` instead.